nexus/src/lib.rs

```diff
@@ -417,6 +418,46 @@ pub async fn run_server(config: &NexusConfig) -> Result<(), String> {
     }
     let internal_server = InternalServer::start(config, &log).await?;
     let server = Server::start(internal_server).await?;
-    server.register_as_producer().await;
     server.wait_for_finish().await
 }
+
+/// Create a new metric producer server.
+fn start_producer_server(
+    log: &Logger,
+    registry: &ProducerRegistry,
+    nexus_addr: SocketAddr,
+) -> Result<ProducerServer, String> {
+    // The producer server should listen on any available port, using the
+    // same IP as the main Dropshot server.
+    let address = SocketAddr::new(nexus_addr.ip(), 0);
+
+    // Create configuration for the server.
+    //
+    // Note that because we're registering with _ourselves_, the listening
+    // address for the producer server and the registration address use the
+    // same IP.
+    let config = oximeter_producer::Config {
+        server_info: ProducerEndpoint {
+            id: registry.producer_id(),
+            kind: ProducerKind::Service,
+            address,
+            base_route: String::from("/metrics/collect"),
+            interval: std::time::Duration::from_secs(10),
+        },
+        // Some(_) here prevents DNS resolution, using our own address to
+        // register.
+        registration_address: Some(nexus_addr),
+        dropshot: dropshot::ConfigDropshot {
+            bind_address: address,
+            request_body_max_bytes: 1024 * 1024 * 10,
```

Comment:
I see sled-agent also uses this value. Is this something producer-specific, or does every producer want the same value? If the latter, should `oximeter_producer::Config` maybe only take `bind_address` and then construct the rest of the `ConfigDropshot` internally? I think `default_handler_task_mode` should also be a decision of `oximeter_producer`, not its users, right?

Reply:
The requests size depends on the data being produced, so I think there still needs to be a knob for it here. The producer server should probably control the handler task mode, though. We could either move the bind address and request size to the `oximeter_producer::Config` directly; or just overwrite the task mode internally? Neither seems awesome.

Reply:
Sounds good, I'll move the configuration parameters around. Thanks!